cpp-client/build-dependencies.sh

```diff
@@ -340,6 +340,8 @@ fi
 #
 : ${GIT_FLAGS:="--quiet -c advice.detachedHead=false"}
 
+BUILD_DIR=build_dir
```

Comment:
I'm not a bash expert, but should this be `export BUILD_DIR=build_dir`?

Reply:
No, is a variable that is only used in the script itself.
You use `export` when you want a variable to get into the shared environment with child processes (so that any processes that your script or shell starts, will ALSO have that variable set, etc).
Without export the variable only applies to the script itself.  Which is exactly what we want.